File: MVVMJsonExample/app/src/main/java/com/blackbirds/shakil/mvvmjsonexample/repositories/MainRepo.java
package com.blackbirds.shakil.mvvmjsonexample.repositories;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import com.blackbirds.shakil.mvvmjsonexample.Common.Common;
import com.blackbirds.shakil.mvvmjsonexample.model.MovieModel;
import com.blackbirds.shakil.mvvmjsonexample.network.APIService;
import com.blackbirds.shakil.mvvmjsonexample.network.Retrofit;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainRepo {
    private static final String TAG = "MainRepo";
    private APIService apiService;

    public MainRepo() {
        this.apiService = Common.getApiService();
    }

    public MutableLiveData<List<MovieModel>> getMovieModelLiveData(){
        MutableLiveData<List<MovieModel>> data = new MutableLiveData<>();
        apiService.getMovieList().enqueue(new Callback<List<MovieModel>>() {
            @Override
            public void onResponse(Call<List<MovieModel>> call, Response<List<MovieModel>> response) {
                Log.e(TAG, "onResponse: " + response.code());
                if (response.isSuccessful()) {
                    data.setValue(response.body());
                } else {
                    data.setValue(null);
                }
            }

            @Override
            public void onFailure(Call<List<MovieModel>> call, Throwable t) {
                Log.e(TAG, "onFailure: " + t.getMessage());
                data.setValue(null);
            }
        });
        return data;
    }
}

File: KotlinMVVMJsonExample/app/src/main/java/com/blackbirds/shakil/kotlinmvvmjsonexample/Repositories/MainRepo.kt
package com.blackbirds.shakil.kotlinmvvmjsonexample.Repositories

import android.util.Log
import androidx.lifecycle.MutableLiveData
import com.blackbirds.shakil.kotlinmvvmjsonexample.Common.Common
import com.blackbirds.shakil.kotlinmvvmjsonexample.Model.MovieModel
import com.blackbirds.shakil.kotlinmvvmjsonexample.Network.APIService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainRepo {
    private val apiService: APIService

    companion object{
        private const val TAG = "MainRepo"
    }

    init {
        apiService = Common.getAPIService
    }

    val getMovieModelLiveData: MutableLiveData<MutableList<MovieModel>>
    get() {
        val data: MutableLiveData<MutableList<MovieModel>> =
            MutableLiveData<MutableList<MovieModel>>()
        apiService.getMovieList().enqueue(object : Callback<MutableList<MovieModel>> {
            override fun onResponse(
                call: Call<MutableList<MovieModel>>,
                response: Response<MutableList<MovieModel>>
            ) {
                Log.e(TAG, "onResponse: " + response.code())
                if (response.isSuccessful){
                    data.value = response.body()
                }
                else{
                    data.value = null
                }
            }

            override fun onFailure(call: Call<MutableList<MovieModel>>, t: Throwable) {
                Log.e(TAG, "onResponse: " + t.message)
                data.value = null
            }

        })
        return data
    }
}